app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
+
+
+    def new
+        @user = User.new
+    end
+
+    def create
+        @user = User.new user_params
+
+        if @user.save
+            session[:user_id] = @user.id
+            redirect_to root_path, notice: "Logged in!"
+        else
+            render :new
+        end
+    end
+
+    def edit
+        @user = User.find(params[:id])
+    end
+
+    def update
+        @user = User.find(params[:id])
+        if @user.update name_params
+            redirect_to root_path, notice: "Updated!"
+        else
+            render :edit
+        end
+    end 
+
```

Comment:
Would of been nice to seen an edit password and update password method in your users controller.